ambry-network/src/main/java/com.github.ambry.network/SSLTransmission.java

```diff
@@ -531,13 +533,12 @@ public int write(ByteBuffer src) throws IOException {
     int written = 0;
     while (src.remaining() != 0) {
       netWriteBuffer.clear();
-      long startTimeMs = SystemTime.getInstance().milliseconds();
+      long startTimeNs = SystemTime.getInstance().nanoseconds();
       SSLEngineResult wrapResult = sslEngine.wrap(src, netWriteBuffer);
-      long encryptionTimeMs = SystemTime.getInstance().milliseconds() - startTimeMs;
-      logger.trace("SSL encryption time: {} ms for {} bytes", encryptionTimeMs, wrapResult.bytesConsumed());
+      long encryptionTimeNs = SystemTime.getInstance().nanoseconds() - startTimeNs;
+      logger.trace("SSL encryption time: {} ns for {} bytes", encryptionTimeNs, wrapResult.bytesConsumed());
       if (wrapResult.bytesConsumed() > 0) {
-        metrics.sslEncryptionTimeInUsPerKB.update(
-            TimeUnit.MILLISECONDS.toMicros(encryptionTimeMs) * 1024 / wrapResult.bytesConsumed());
+        metrics.sslEncryptionTimeInUsPerKB.update(encryptionTimeNs / wrapResult.bytesConsumed());
```

Comment:
sslEncryptionTimeInUsPerKB = Us/KB = (ns/1000) / (bytes/1024) = ns / bytes. Looks good, no?

Reply:
I think it's only roughly equivalent if you use KiB, since its actually 1 Us/KiB = 1000/1024 ns/bytes.

If you use power of 10 KB, then they would be equivalent